[12.0][MIG] invoice_default_account_date >> account_invoice_default_account_date

vvrossem opened the pull request:
[Task](https://gestion.coopiteasy.be/web#id=5503&view_type=form&model=project.task&action=475&active_id=162)
Now renamed `account_invoice_default_account_date`
Tested and validated

robinkeunen committed:
[IMP] if unset set invoice `date` to today on write

vvrossem committed:
[MIG] invoice_default_account_date: migration to 10.0

vvrossem committed:
[MIG] invoice_default_account_date: migration to 11.0

vvrossem committed:
[MIG] invoice_default_account_date: migration to 12.0

vvrossem committed:
[IMP] invoice_default_account_date: black formatter

vvrossem committed:
[IMP] invoice_default_account_date: author and website

vvrossem committed:
[IMP] invoice_default_account_date: readme

vvrossem committed:
[REF] renamed invoice_default_account_date > account_invoice_*

vvrossem committed:
[IMP] account_invoice_default_account_date: pre-commit

vvrossem committed:
[UPD] account_invoice_default_account_date: readme

vvrossem commented:
@robinkeunen could you kindly review this PR ? I'll need to merge it before Sun 22/11 :scream:

robinkeunen commented on account_invoice_default_account_date/models/account_invoice.py:
```diff
@@ -0,0 +1,20 @@
+        # vendor bills
+        if (
+            not self.date
+            and not vals.get("date")
```
The python way: `and "date" not in vals`

vvrossem replied:
done

robinkeunen commented on account_invoice_default_account_date/models/account_invoice.py:
```diff
@@ -0,0 +1,20 @@
+            and self.type in ("in_invoice", "in_refund")
+        ):
+
+            vals["date"] = date.today()
```
Prefer using

```py
from odoo.models import Date
vals["date"] = Date.context_today()
```

It will localize the date. It could matter if the timezone difference sends you across midnight.

vvrossem replied:
changed with : 
```python
from odoo import fields
vals["date"] = fields.Date.context_today(self)
```

robinkeunen reviewed (changes requested):
What this will do is set the `date` as today() at any write (even if still in draft). Don't you want to set the date either when the invoice is validated or switches to 'open'?

vvrossem committed:
[REF] account_invoice_default_account_date: fields.Date
- `fields.Date.context_today(self)` instead of `date.today()`
- `"date" not in vals` instead of `not vals.get("date")`

vvrossem commented:
> Don't you want to set the date either when the invoice is validated or switches to 'open'?

@robinkeunen I've only ported the module to 12.0 :) Since it has already been tested and validated, I propose we refactor this module on another occasion

robinkeunen reviewed (approved):
Ok then can you add a note in the Known Issues and/or a comment in the code?

👍

vvrossem commented:
@robinkeunen aaaaaaaaaaaaaaaaaaaaaaaaaaaaand it's done! :tada:

vvrossem commented:
reset to previous commit since `write` is called on invoice validation

vvrossem committed:
[ADD] account_invoice_default_account_date: roadmap.rst
- updated readme.rst
